cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/VolumeStorage.kt

```diff
@@ -1,24 +1,27 @@
 package com.cognifide.cogboard.storage
 
-import com.cognifide.cogboard.config.validation.Validator
+import com.cognifide.cogboard.CogboardConstants
 import com.cognifide.cogboard.config.ConfigType
 import io.vertx.core.json.JsonObject
 import io.vertx.core.logging.Logger
 import io.vertx.core.logging.LoggerFactory
 import java.io.File
 
 class VolumeStorage(
-    private val configType: ConfigType,
-    private val path: String,
-    private val validator: Validator
+    private val type: ConfigType,
+    private val configFile: String = type.configFile()
 ) : Storage {
 
-    override fun loadConfig() = loadConfig(configType, path, validator)
+    override fun loadConfig(): JsonObject {
+            val conf = File(configFile).readText()
+            return if (type.validator.validate(conf)) JsonObject(conf)
+            else CogboardConstants.errorResponse("$type config not valid")
+        }
 
     override fun saveConfig(configJson: JsonObject): Boolean {
         val conf = configJson.toString()
-        if (validator.validate(conf)) {
-            File(path).writeText(conf)
+        if (type.validator.validate(conf)) {
```

Comment:
You break the Law of Demeter twice : here and on the line 17.
I would suggest extracting the validating method to the ConfigType enum.

Reply:
Done, THX for the tip somehow I've missed it